libp2p/src/main/kotlin/io/libp2p/pubsub/gossip/GossipRouter.kt

```diff
@@ -416,15 +416,18 @@ open class GossipRouter(
                     }
                     .flatten()
             }
-        val list = peers
-            .filterNot { peerDoesNotWantMessage(it, msg.messageId) }
-            .map { submitPublishMessage(it, msg) }
 
         mCache += msg
-        flushAllPending()
 
-        return if (list.isNotEmpty()) {
-            anyComplete(list)
+        return if (peers.isNotEmpty()) {
+            val publishedMessages = peers
+                .filterNot { peerDoesNotWantMessage(it, msg.messageId) }
+                .map { submitPublishMessage(it, msg) }
+            if (publishedMessages.isEmpty()) {
+                return CompletableFuture.completedFuture(Unit)
+            }
+            flushAllPending()
+            anyComplete(publishedMessages)
```

Comment:
nit: Could you please just rearrange this code a bit: 
```suggestion
            if (publishedMessages.isEmpty()) {
                CompletableFuture.completedFuture(Unit)
            } else {
                flushAllPending()
                anyComplete(publishedMessages)
            }
```

Reply:
Done